interfaces/apparmor/backend.go

```diff
@@ -586,6 +586,18 @@ func (b *Backend) SetupMany(snaps []*snap.Info, confinement func(snapName string
 	return errors
 }
 
+func RemoveAllSnapAppArmorProfiles() error {
+	dir := dirs.SnapAppArmorDir
+	globs := []string{"snap.*", "snap-confine.*"}
+	cache := apparmor_sandbox.CacheDir
+	_, removed, errEnsure := osutil.EnsureDirStateGlobs(dir, globs, nil)
+	errUnload := unloadProfiles(removed, cache)
```

Comment:
do we really want to unload them as well? doesn't that mean that we are without profiles for a while? I vaguely thought that the idea was to remove them from disk but keep them in memory until they can be re-generated?

Reply:
Ah that is a good point - otherwise they are potentially unconfined too during this time.

Reply:
Heh actually looking at the implementation of `UnloadProfiles()` it doesn't actually unload the profile from the kernel currently (https://github.com/snapcore/snapd/blob/fe362e6374dd2c528aecac34b7f3908a2c2c7a4f/sandbox/apparmor/profile.go#L127) - but it does remove them from the profile cache - which I think it perhaps still worth doing. So either I can just remove the commented-out code that actually unloads the profiles from the kernel and rename this function + refactor all the callers, or we split it up so that we can still say remove the profile from the cache without actually doing the unloading and then this is what can be called by `RemoveAllSnapAppArmorProfiles()`.

I prefer this second option but am happy with whatever you want @pedronis - let me know.

Reply:
looking at the comment there, the precondition will never be totally true because of things like enduring services, so I'm tempted to actually go for the rename so that the name and behavior match. You could leave some TODO about considering unloading again but as I said we do have cases where we can't be sure no processes are running so it feels inherently fragile.

Reply:
I agree - done in https://github.com/snapcore/snapd/pull/12543/commits/b1f7eabff805adefe78210d758d5023643da203c